### PR TITLE
Music/Sound component

### PR DESCRIPTION
**Pull request by TrueBabyChaise**
## Proposed changes

Add music and sound components with their respective system to handle audio.

## Information
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Non-breaking changes

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

### CODE REVIEW

File: sources/Runner/Runner.cpp [hunk 1]
```diff
@@ -56,12 +60,14 @@ namespace BBM
...
-			.addSystem<MovableSystem>();
+			.addSystem<MovableSystem>()
+			.addSystem<SoundManagerSystem>()
+			.addSystem<MusicSystem>();
 	}
 
 	void enableRaylib(WAL::Wal &wal)
 	{
-		RAY::TraceLog::setLevel(LOG_WARNING);
+		//RAY::TraceLog::setLevel(LOG_WARNING);
```
Comment (Arthi-chaud) [inline sources/Runner/Runner.cpp:70]: This should not be commented

**Review by Arthi-chaud — CHANGES_REQUESTED** [review]
The fact that you have to give an entire map, some fields being empty, really doesn't make it flexible

File: sources/Component/Sound/SoundComponent.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+#include <iostream>
+#include <memory>
+#include "SoundComponent.hpp"
+
+namespace BBM
+{	
+	float SoundComponent::volume = 0.75;
+
+	SoundComponent::SoundComponent(WAL::Entity &entity, \
```
Comment (AnonymusRaccoon) [inline sources/Component/Sound/SoundComponent.cpp:13]: Please don't use \, simply allign your second argument.

File: sources/Component/Sound/SoundComponent.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+		this->_soundList[this->_soundIndex].get()->pause();
+	}
+
+	void SoundComponent::setVolume(float &volumeUpdate)
+	{
+		if (!this->_isLoad.at(this->_soundIndex))
+			return;
+		if (volumeUpdate >= 0)
+			this->volume = volumeUpdate;
+			this->_soundList[this->_soundIndex].get()->setVolume(this->volume);
```
Comment (AnonymusRaccoon) [inline sources/Component/Sound/SoundComponent.cpp:63]: I believe you wanted this on the if.

File: sources/Component/Sound/SoundComponent.hpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace BBM
+{
+	//! @brief A basic Sound component
+	class SoundComponent : public WAL::Component
+	{
+	public:
+
+		//! @brief All sounds of the player
+		enum soundIndex {
```
Comment (AnonymusRaccoon) [inline sources/Component/Sound/SoundComponent.hpp:19]: Use PascalCase for type names.

**Review by Arthi-chaud — APPROVED** [review]

**Review by EternalRat — APPROVED** [review]

**Review by GitBluub — APPROVED** [review]

**Review by AnonymusRaccoon — APPROVED** [review]

**Pull request merged**